src/components/Unknown/BuildScreen/Builder/CPUBuilder/index.tsx

```diff
@@ -52,14 +36,14 @@ const CPUBuilder: React.FC = () => {
     setExpand((prev) => !prev);
   };
 
-  const products = parts.map((part) => (
-    <BuilderProduct
-      product={part}
-      key={part.id}
-      handleSelect={handleAddProduct}
-      selectedId={selectedId}
-    />
-  ));
+  const firestore = useFirestore().collection('CPUs');
+  const { data, status } = useFirestoreCollectionData<CPU>(firestore);
+
+  useEffect(() => {
+    if (status === 'success') {
+      setCPUs(normalizeCPUs(data));
+    }
+  }, [data, status]);
```

Comment:
this is unnecessary as well as CPUs state
 const { data: CPUs, status } = useFirestoreCollectionData<CPU>(firestore);

Reply:
agree